tools-v2/pkg/cli/command/curvebs/update/peer/peer.go

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveCli
+ * Created Date: 2023-02-27
+ * Author: zweix
+ */
+
+package peer
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/cli2"
+	"github.com/opencurve/curve/tools-v2/proto/proto/common"
+)
+
+const (
+	updateExample = `$ curve bs update peer 127.0.0.1:8080:0 --logicalPoolId=1 --copysetId=10001 \
+ --rpcretrytimes=1 --rpctimeout=10s`
+)
+
+type ResetRpc struct {
+	Info    *basecmd.Rpc
+	Request *cli2.ResetPeerRequest2
+	Client  cli2.CliService2Client
+}
+
+func (rRpc *ResetRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	rRpc.Client = cli2.NewCliService2Client(cc)
+}
+
+func (rRpc *ResetRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return rRpc.Client.ResetPeer(ctx, rRpc.Request)
+}
+
+// Options rpc request options.
+type Options struct {
+	Timeout    time.Duration
+	RetryTimes int32
+}
+
+type PeerCommand struct {
+	basecmd.FinalCurveCmd
+
+	// request parameters
+	opts          Options
+	logicalPoolID uint32
+	copysetID     uint32
+
+	resetPeer *common.Peer
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*PeerCommand)(nil) // check interface
+
+// NewCommand ...
+func NewPeerCommand() *cobra.Command {
+	peerCmd := &PeerCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "peer",
+			Short:   "update(reset) the peer from the copyset",
+			Example: updateExample,
+		},
+	}
+	basecmd.NewFinalCurveCli(&peerCmd.FinalCurveCmd, peerCmd)
+	return peerCmd.Cmd
+}
+
+func (pCmd *PeerCommand) AddFlags() {
+	config.AddRpcRetryTimesFlag(pCmd.Cmd)
+	config.AddRpcTimeoutFlag(pCmd.Cmd)
+
+	config.AddBSLogicalPoolIdFlag(pCmd.Cmd)
+	config.AddBSCopysetIdFlag(pCmd.Cmd)
+}
+
+// ParsePeer parse the peer string
+func ParsePeer(peer string) (*common.Peer, *cmderror.CmdError) {
+	cs := strings.Split(peer, ":")
+	if len(cs) != 3 {
+		pErr := cmderror.ErrGetPeer()
+		pErr.Format(fmt.Sprintf("error format for the peer info %s", peer))
```

Comment:
here it can be written like this:
```suggestion
	pErr := cmderror.ErrSplitPeer ()
    pErr.Format(peer)
```
and
```suggestion
	ErrSplitPeer = func() *CmdError {
		return NewInternalCmdError(15, "split peer[%s] failed, peer should be like: 127.0.0.1:6700:0")
	}
```